servant-server/src/Servant/Server/Internal.hs

```diff
@@ -351,12 +361,23 @@ instance (KnownSymbol sym, FromHttpApiData a, HasServer api context)
         -- named "foo" or "foo[]" and call parseQueryParam on the
         -- corresponding values
         parameters r = filter looksLikeParam (querytext r)
-        values r = mapMaybe (convert . snd) (parameters r)
-    in  route (Proxy :: Proxy api) context (passToServer subserver values)
+        parseParam (paramName, paramTxt) =
+          case parseQueryParam (fromMaybe "" paramTxt) of
```

Comment:
I would at least not change previously existing behaviour here (which was to ignore params without values). If this is a concern we can solve it in a different issue/PR.